lib/tpm2.c

```diff
@@ -591,13 +592,36 @@ tool_rc tpm2_evictcontrol(ESYS_CONTEXT *esys_context,
     return tool_rc_success;
 }
 
+uint32_t fix_esys_hierarchy(TPMI_RH_HIERARCHY hierarchy)
```

Comment:
Why a uint32_t and not one of the other typedefs?

Reply:
Because it can return either TPMI_RH_HIERARCHY or ESYS_TR, which are both typedefs of uint32_t anyway.